### DISCUSSION TITLE
Sergen always generates code using namespaces

### DISCUSSION THREAD

**Discussion by dfaruque** [serenity-is/Serenity#6685] [Help]
I have created a new project using latest Serene template. Everything works good with modular typescript. But sergen still generates code using namespace.

I have tried to set `"ModuleTypings": true` in sergen.json but no luck.
```
  "ServerTypings": {
    "ModuleTypings": true
  },
```

What am I missing?

**Comment by wezmag** [comment]
Did you check sergen's version?
``` cli
dotnet sergen -v
```
If the sergen is not the latest version, try upgrade it.
``` cli
dotnet tool update sergen
``` 
You should see something like below.
```
Tool 'sergen' is up to date (version '6.5.1' manifest file D:\Source\repos\Serene8\Serene8\Serene8.Web\.config\dotnet-tools.json) .
```
*Run the commands in `YourProject.Web` folder.

**Reply by dfaruque** [reply]
sergen is up-to-date. but no luck.

**Reply by wezmag** [reply]
Can you show the content of `tsconfig.json`?

When I change the value of `compilerOptions`>`module` from `ESNext` to `none`, sergen start generating typescript files with namespace.

sergen also checks if there is another `tsconfig.json` in the Namespaces folder.
https://github.com/serenity-is/Serenity/blob/19d63d1fe0bc4533e4e3bfd1732b86e52112560d/src/Serenity.Net.CodeGenerator/TypeScript/TSConfigHelper.cs#L229

**Reply by dfaruque** [reply]
Thank you @wezmag for your time and cooperation. 

I have not change anything in tsconfig.json. it is fresh new project created with latest serene template.
However, here is my tsconfig.json

```
{
  "compileOnSave": true,
  "compilerOptions": {
    "allowJs": false,
    "baseUrl": ".",
    "declaration": false,
    "emitBOM": true,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": true,
    "jsx": "react",
    "module": "ESNext",
    "moduleResolution": "Node",
    "noEmit": true,
    "noEmitHelpers": false,
    "noEmitOnError": false,
    "outDir": "./wwwroot/esm",
    "paths": {
      "@/*": [ "Modules/*" ],
      "@serenity-is/*": [ "node_modules/@serenity-is/*/dist/index" ]
    },
    "preserveConstEnums": true,
    "rootDir": ".",
    "skipDefaultLibCheck": true,
    "skipLibCheck": true,
    "sourceMap": false,
    "target": "ES6",
    "typeRoots": [
      "./node_modules/@types"
    ],
    "types": [
      "jquery",
      "jquery.validation",
      "jqueryui",
      "toastr"
    ]
  },
  "include": [
    "./Modules/**/*"
  ],
  "typeAcquisition": {
    "enable": false
  }
}
```
